cli/Sources/TuistCI/CIController.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import Mockable
+import TuistSupport
+
+@Mockable
+public protocol CIControlling {
+    /// Detects CI environment and returns CI information
+    func ciInfo() -> CIInfo?
+}
+
+public struct CIController: CIControlling {
+    public init() {}
+
+    public func ciInfo() -> CIInfo? {
+        let env = Environment.current.variables
+
+        // GitHub Actions
+        if env["GITHUB_ACTIONS"] != nil {
```

Comment:
Should we also detect if the runners are provided by a third-party provider (e.g. Namespace), and link to them? It'd be subtle detail that they'll appreciate.